script.js

```diff
@@ -1,110 +1,175 @@
+  image.src = episode.image ? episode.image.medium : './No-Image-Placeholder.png';
 
-  episodeCard.querySelector(".episode-card__title")
-    .innerText = episodeTitle;
+  const summary = document.createElement('p');
+  summary.classList.add('episode-card__summary');
+  summary.innerHTML = episode.summary;
 
-  episodeCard.querySelector(".episode-card__img").src = episodeImage || './No-Image-Placeholder.png';
-  episodeCard.querySelector(".episode-card__summary").outerHTML = episodeSummary;
+  // We're putting all the information about the episode together on the card.
+  dataContainer.appendChild(image);
+  dataContainer.appendChild(summary);
+  episodeCard.appendChild(title);
+  episodeCard.appendChild(dataContainer);
 
   return episodeCard;
 }
 
-function setupSearch() {
-  const searchInput = document.getElementById("search-input");
-  searchInput.addEventListener("input", function () {
-    const searchTerm = searchInput.value.trim().toLowerCase();
-    filterEpisodes(searchTerm);
+// We're making a menu of episode numbers so people can pick a specific episode to watch.
+function populateEpisodeNumberFilter(episodeList) {
+  const episodeNumberFilterElement = document.getElementById('episode-number-filter');
+
+  // We're clearing out any old options from the menu.
+  episodeNumberFilterElement.innerHTML = "";
+
+  // We're making an option to show all episodes in case people want to see them all.
+  const defaultOption = document.createElement('option');
+  defaultOption.value = 'all';
+  defaultOption.textContent = 'Show All Episodes';
+  episodeNumberFilterElement.appendChild(defaultOption);
+
+  // We're adding each episode number to the menu so people can pick one.
+  episodeList.forEach(episode => {
+      const optionElement = document.createElement('option');
+      optionElement.value = `S${episode.season.toString().padStart(2, '0')}E${episode.number.toString().padStart(2, '0')}`;
+      optionElement.textContent = `${optionElement.value} - ${episode.name}`;
+      episodeNumberFilterElement.appendChild(optionElement);
   });
-}
-
-function filterEpisodes(searchTerm) {
-  const filteredEpisodes = allEpisodes.filter(
-    (episode) =>
-      episode.name.toLowerCase().includes(searchTerm) ||
-      episode.summary.toLowerCase().includes(searchTerm)
-  );
 
-  makePageForEpisodes(filteredEpisodes);
+  // When someone picks an episode from the menu, we want to show that episode and hide the others.
+  episodeNumberFilterElement.addEventListener('change', function () {
+      const selectedEpisode = this.value;
+      filterEpisodesByEpisodeNumber(selectedEpisode, episodeList);
+  });
 }
 
-function updateEpisodeListCounter(episodeList) {
-  const episodesDisplayAmount = document.querySelector(
-    ".episodes-display-amount"
-  );
-  episodesDisplayAmount.textContent = `Displaying ${episodeList.length} out of ${allEpisodes.length} episodes`;
```

Comment:
I think the episode counter is necessary since it's shown in the steps and screenshots.